code/clue.js

```diff
@@ -1,59 +1,361 @@
+      document.getElementById('killerDescription').innerHTML = (`Description: ${mystery.killer.description}`);
+      killerPicked = true;
+      checkScroll();
+    }, 2000);
+  }
+}
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+const pickWeapon = (loadID) => {
+  if (killerPicked) {
+    const weaponsIncreased = increaseChanceForFavWeapon(mystery.killer);
+    document.getElementById(loadID.id).style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById(loadID.id).style.opacity = 0;
+      mystery.weapon = randomSelector(weaponsIncreased);
+      document.getElementById('weaponName').style.background = mystery.weapon.color;
+      document.getElementById('weaponName').innerHTML = `Type: ${mystery.weapon.name}`;
+      document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`;
+      weaponPicked = true;
+      checkScroll();
+    }, 2000);
+  } else if (showInfoMessage("pickWeapon"));
+}
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+const pickRoom = (loadID) => {
+  if (killerPicked && weaponPicked) {
+    document.getElementById(loadID.id).style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById(loadID.id).style.opacity = 0;
+      const increasedRoom = increaseChanceForRoom(mystery.weapon);
+      mystery.room = randomSelector(increasedRoom);
+      document.getElementById('roomName').innerHTML = mystery.room;
+      roomPicked = true;
+      checkScroll();
+    }, 2000);
+  } else(showInfoMessage("pickRoom"));
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const showInfoMessage = (callerFunctionName) => {
+  if (callerFunctionName === "pickWeapon" || (callerFunctionName === "pickRoom" && !killerPicked)) {
+    document.getElementById("killerInfoMessage").style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById("killerInfoMessage").style.opacity = 0;
+    }, 2000);
+  } else if (callerFunctionName === "pickRoom" && killerPicked) {
+    document.getElementById("weaponInfoMessage").style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById("weaponInfoMessage").style.opacity = 0;
+    }, 2000);
+  } else if (callerFunctionName === "revealMystery") {
+    document.getElementById("mysteryInfoMessage").style.opacity = 1;
+    setTimeout(function () {
+      document.getElementById("mysteryInfoMessage").style.opacity = 0;
+    }, 2000);
+  }
```

Comment:
Some comments could help a lot at first glance here.

Also, consider simplifying this function by creating a `displayHint` function.  Maybe something like

```javascript
const displayHint = (hintId) =>{
  document.getElementById(hintId).style.opacity = 1;
  setTimeout(function () {
      document.getElementById(hintId).style.opacity = 0;
    }, 2000);
}
```

It might make this function more readable